### PR TITLE
V2: Replace `MethodUnaryDescriptor` with `DescMethodUnary` from `@bufbuild/protobuf`

### PR DESCRIPTION
**Pull request by srikrsna-buf**
Replace `MethodUnaryDescriptor` with `DescMethodUnary` from `@bufbuild/protobuf`.

### CODE REVIEW

File: examples/react/basic/package.json [hunk 1]
```diff
@@ -26,10 +26,10 @@
     "@testing-library/react": "^16.0.1",
     "@types/react": "^18.3.5",
     "@types/react-dom": "^18.3.0",
-    "@vitejs/plugin-react": "^4.3.1",
+    "@vitejs/plugin-react": "^4.3.2",
```
Comment (paul-sachs) [inline examples/react/basic/package.json:29]: Did we mean to update these dependencies here? Are they somehow related to protobuf updates?

File: packages/connect-query/src/create-query-options.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ export function createQueryOptions<
   queryKeyHashFn: (queryKey: QueryKey) => string;
 } {
   const queryKey = createConnectQueryKey({
-    schema,
+    schema: schema,
```
Comment (paul-sachs) [inline packages/connect-query/src/create-query-options.ts:72]: Unnecessary change.

File: examples/react/basic/package.json (hunk 1 above)
Comment (srikrsna-buf) [reply examples/react/basic/package.json:29]: Ah yes sorry, I forgot to add context. When I tried to update protobuf version, pnpm was downloading types from two different vite versions, which was resulting in a type error in the example's vite config. I tried the usual steps of deleting the lockfile and removing node modules didn't work. So I upgraded to latest.

Comment (paul-sachs) [reply examples/react/basic/package.json:29]: Makes sense.

**Review by paul-sachs — APPROVED** [review]

**Comment by timostamm** [comment]
Updating the docs in https://github.com/connectrpc/connect-query-es/pull/449/commits/f7c95a2648d8b41b9db430cefceec7e755f1a089.
